src/parser/mod.rs

```diff
@@ -11173,15 +11173,42 @@ impl<'a> Parser<'a> {
                     self.peek_token().location
                 )
             }
-            expr => self
-                .parse_optional_alias(keywords::RESERVED_FOR_COLUMN_ALIAS)
-                .map(|alias| match alias {
-                    Some(alias) => SelectItem::ExprWithAlias { expr, alias },
-                    None => SelectItem::UnnamedExpr(expr),
-                }),
+            expr => {
+                if dialect_of!(self is MsSqlDialect) {
```

Comment:
We recently started to discourage the use of [dialect_of](https://github.com/apache/datafusion-sqlparser-rs/blob/383226c3e81d9e5ed64f6f188815dfc9d7fea317/src/dialect/mod.rs#L63-L72) can we update this to use a method on the Dialect trait as described in the link?
We could add a method like `supports_eq_alias_assigment` or similar